### ISSUE TITLE
[css-display-3] text run ?

### ISSUE THREAD

**Issue by letochagone** [w3c/csswg-drafts#7768]
in the second paragraph , https://www.w3.org/TR/css-display-3/#intro
what does the term "text run" mean?  : 

> Each box in the [box tree](https://www.w3.org/TR/css-display-3/#box-tree) represents its corresponding [element](https://www.w3.org/TR/css-display-3/#elements) (or [pseudo-element](https://www.w3.org/TR/selectors-4/#pseudo-element))
>  in space and/or time on the canvas, while each **_TEXT RUN_** in the box tree likewise represents 
> the contents of its corresponding [text nodes](https://www.w3.org/TR/css-display-3/#text-nodes).

after a lot of time trying to translate this term into my language, I think it may be a mistake and that this word "run" was added by mistake.

**Comment by myakura** [comment]
I don't think that's added by mistake. While it's a different spec, HTML also uses "run" as a noun meaning like a sequence.
(And that confuses people, too. https://github.com/whatwg/html/issues/6782)

**Comment by Loirooriol** [comment]
Yeah, a text run is the thing in the box tree that represents a sequence of adjacent DOM text nodes in the element tree. Basically:

 - The DOM tree contains element nodes, text nodes, comment nodes, doctype nodes, etc.
 - The element tree contains elements, pseudo-elements, and text nodes.
 - The box tree contains boxes (representing elements or pseudo-elements) and text runs (representing sequences of text nodes).

**Comment by fantasai** [comment]
Re-opening in response to @myakura's comment, because maybe we should rename “text run” to “text sequence”!

**Comment by dbaron** [comment]
"text run" is a term used elsewhere for similar purposes, e.g. https://learn.microsoft.com/en-us/dotnet/api/system.windows.media.textformatting.textrun?view=windowsdesktop-7.0  That said, I'm not sure if the use in the display spec matches the meaning in Gecko (where it has been used for a long time).

**Comment by astearns** [comment]
I think “text run” is a term of art that does not quite match what is being described in the spec, and I don’t see a benefit in using that term that offsets the confusion being generated. So I support changing to “text sequence.”

**Comment by astearns** [comment]
Since this change is fairly low-impact we likely do not need to spend meeting time on it.

The CSSWG will automatically accept this resolution one week from now if no objections are raised here. Anyone can add an emoji to this comment to express support. If you do not support this resolution, please add a new comment.

Proposed Resolution: Change the term “text run” to “text sequence” in css-display-3

**Comment by astearns** [comment]
RESOLVED: Change the term “text run” to “text sequence” in css-display-3

**Comment by myakura** [comment]
There are other specs that use "text run" as well. I see Text and Writing Modes use it. Does the resolution cover other specs?
https://drafts.csswg.org/css-text-4/#valdef-word-boundary-detection-normal
https://drafts.csswg.org/css-writing-modes-4/#text-combine-runs

**Comment by astearns** [comment]
We would definitely need to change css-text since it links to the css-display definition. I think we should also change css-writing-modes for consistency.

**Comment by fantasai** [comment]
Yes, we would apply the change across all the CSS specs that reference the same concept.

**Comment by fantasai** [comment]
@letochagone Ok, I've updated the editor's draft in https://github.com/w3c/csswg-drafts/commit/e0f9cb9e839f6b737aa3a5169b0b791756096a4e to use the word “text sequence”; hopefully this will be less confusing. :) Let me know what you think.

@myakura I also updated all of the other specs that seemed to be referencing the same concept. There remain some uses of “text run” in Writing Modes, Text, and Fonts which don't seem to be quite tied to the same box tree concept, so I left those alone. Feel free to file a follow-up issue if you think something is off. CC @svgeesus

**Comment by Loirooriol** [comment]
@fantasai Why do flex and grid use different wordings now?

> and each contiguous sequence of child text sequences

> and each child <a>text sequence</a>

I think the later is better, "sequence of sequences" seems unnecessarily verbose.

**Comment by fantasai** [comment]
@Loirooriol Fixed, thanks.

**Comment by SelenIT** [comment]
> I think the latter is better, "sequence of sequences" seems unnecessarily verbose.

But doesn't it affect the meaning? Will "each child text sequence" not imply that each text sequence created by a `display:contents` child become a separate flex/grid item? I believe that "sequence of sequences" wording was specifically introduced to clarify that they will become a single flex/grid item (possibly with differently styled parts of text), though I agree that there should be better wording to express this intent.

**Comment by Loirooriol** [comment]
Good point, I didn't consider `display: contents`.

**Comment by letochagone** [comment]
@fantasai Perfect! It's much clearer now. Thank you!

**Comment by Loirooriol** [comment]
Not perfect, it's wrong, see above

**Comment by ByteEater-pl** [comment]
The term *text run* is also used in SVG.

**Comment by letochagone** [comment]
I found another text where this "untranslatable" word is used:
https://html.spec.whatwg.org/multipage/dom.html#phrasing-content

<img width="807" height="103" alt="Image" src="https://github.com/user-attachments/assets/ffc7033d-55d2-4ca5-8de9-873a33b524ef" />

**Comment by SelenIT** [comment]
Returning to the topic, I'd like to clarify the following piece of the current definition:

> Similarly, each contiguous sequence of sibling [text nodes](https://drafts.csswg.org/css-display/#text-nodes) generates a [text sequence](https://drafts.csswg.org/css-display/#css-text-sequence) containing their text contents, _which is assigned the same styles as the generating text nodes._

The part I emphasized seems a bit ambiguous to me: does it already imply that a single text sequence can contain parts of text with multiple different stylings (e.g. in case of `display:contents`) or not? If yes, the change above appears correct, but a separate term for each same-style sub-sequence would probably be needed. IIRC, the original "text run" term referred to the same-style sequences, and the "sequence of sequences" wording intended to highlight it.
